types/denom.go

```diff
@@ -60,5 +75,70 @@ func ConvertCoin(coin Coin, denom string) (Coin, error) {
 		return NewCoin(denom, coin.Amount), nil
 	}
 
-	return NewCoin(denom, coin.Amount.ToDec().Mul(srcUnit.Quo(dstUnit)).TruncateInt()), nil
+	return NewCoin(denom, coin.Amount.ToDec().Mul(srcUnit).Quo(dstUnit).TruncateInt()), nil
+}
+
+// ConvertDecCoin attempts to convert a decimal coin to a given denomination. If the given
+// denomination is invalid or if neither denomination is registered, an error
+// is returned.
+func ConvertDecCoin(coin DecCoin, denom string) (DecCoin, error) {
```

Comment:
Would like to see unit tests for this function (similar to what's done for ConvertCoin).

Reply:
done